src/main/java/com/softserve/academy/spaced/repetition/service/CourseCommentService.java

```diff
@@ -59,6 +71,32 @@ public CourseComment updateCommentById(Long commentId, String commentText) {
     }
 
     public void deleteCommentById(Long commentId) {
+        LOGGER.debug("Deleted comment with id:{}", commentId);
+        if (commentRepository.findOne(commentId).getParentCommentId() == null) {
+            commentRepository.findCourseCommentsByParentCommentId(commentId).stream()
+                    .forEach(comment -> commentRepository.delete(comment.getId()));
+        }
         commentRepository.delete(commentId);
     }
+
+    public List<CommentDTO> getTreeOfComments(List<CommentDTO> comments){
```

Comment:
Please move this method to separate class on controller level. You can even name it like CommentTreeBuilder. Service level should not be aware of any ***DTO class.